### PR TITLE
Add exercise 4

### PR DESCRIPTION
**Pull request by bkdnntdat**

### CODE REVIEW

File: src/main/java/PTTSNTVer2.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+public class PTTSNTVer2 {
+    public String PhanTich(long a){
```
Comment (voquanghoa) [inline src/main/java/PTTSNTVer2.java]: Sai name convention

File: src/main/java/PTTSNTVer2.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        String s="";
```
Comment (voquanghoa) [inline src/main/java/PTTSNTVer2.java]: Bad name `s`

File: src/main/java/PTThuaSoNguyenTo.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+public class PTThuaSoNguyenTo {
+    public String PhanTich(long a){
```
Comment (voquanghoa) [inline src/main/java/PTThuaSoNguyenTo.java]: Sai convention

File: src/main/java/PTTSNTVer2.java (hunk 1 above)
Comment (bkdnntdat) [reply src/main/java/PTTSNTVer2.java]: Dạ em sửa r ạ

File: src/main/java/PTTSNTVer2.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            while(a%i==0){
+                dem++;
+                a/=i;
+            }
+            if (dem==1) s+= i + " * ";
+            else {
+                if (dem > 1) s += i + "^" + dem + " * ";
+            }
+        }
+        s=s.substring(0,s.length()-3);
```
Comment (bkdnntdat) [inline src/main/java/PTTSNTVer2.java]:
```suggestion
        s=s.substring(0, s.length()-3);
```

**Pull request merged**